tests/test-sessions/test-jdbc-sessions/src/test/java/org/eclipse/jetty/server/session/JdbcTestHelper.java

```diff
@@ -84,14 +129,34 @@ public static void shutdown(String connectionUrl)
         }
     }
 
+    public static DatabaseAdaptor buildDatabaseAdaptor()
+    {
+        DatabaseAdaptor da = new DatabaseAdaptor();
+        if (USE_MARIADB)
+        {
+            da.setUsername(MARIAD_DB.getUsername());
```

Comment:
How would the mariadb instance have a username and password set?? This code doesn't pass in any username or password to the MariaDBContainer, so it can possibly be set can it?

Reply:
this is part of the features provided by the extension of `JdbcDatabaseContainer`. By default MariaDBContainer have default username/password but if you prefer we can use our own. There are methods for that to call before starting the container. The `MariaDBContainer` just the image with env vars `MYSQL_USER` and `MYSQL_PASSWORD` see https://hub.docker.com/_/mariadb

Reply:
The 2 ways to use the DatabaseAdaptor are to pass it a DataSource or a DriverInfo - can't mariadb use one of these methods?